builder/export.go

```diff
@@ -0,0 +1,227 @@
+package builder
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/bitrise-io/go-utils/pathutil"
+)
+
+// FileInfoWithPath ...
+type FileInfoWithPath struct {
+	modTime time.Time
+	path    string
+}
+
+func fileInfos(dir string) ([]FileInfoWithPath, error) {
+	fileInfos := []FileInfoWithPath{}
+
+	if err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		fileInfos = append(fileInfos, FileInfoWithPath{info.ModTime(), path})
+
+		return nil
+	}); err != nil {
+		return []FileInfoWithPath{}, err
+	}
+
+	return fileInfos, nil
+}
+
+func isInTimeInterval(t, startTime, endTime time.Time) bool {
+	if startTime.After(endTime) {
+		return false
+	}
+	return (t.After(startTime) || t.Equal(startTime)) && (t.Before(endTime) || t.Equal(endTime))
+}
+
+func filterFilesInfosForTimeWindow(fileInfos []FileInfoWithPath, startTime, endTime time.Time) []FileInfoWithPath {
+	if startTime.IsZero() || endTime.IsZero() || startTime.Equal(endTime) {
+		return []FileInfoWithPath{}
+	}
+	if startTime.After(endTime) {
+		return []FileInfoWithPath{}
+	}
+
+	filteredFileInfos := []FileInfoWithPath{}
+
+	for _, fileInfo := range fileInfos {
+		if isInTimeInterval(fileInfo.modTime, startTime, endTime) {
+			filteredFileInfos = append(filteredFileInfos, fileInfo)
+		}
+	}
+
+	return filteredFileInfos
+}
+
+// finds the last modified file matching to most strict regexp
+// order of regexps should be: most strict -> less strict
+func findLastModifiedWithFileNameRegexps(fileInfos []FileInfoWithPath, regexps ...*regexp.Regexp) *FileInfoWithPath {
+	if len(fileInfos) == 0 {
+		return nil
+	}
+
+	lastModifiedFileInfoPtr := new(FileInfoWithPath)
+
+	if len(regexps) > 0 {
+		for _, re := range regexps {
+			for _, fileInfo := range fileInfos {
+				fileName := filepath.Base(fileInfo.path)
+				if match := re.FindString(fileName); match == fileName {
+					if lastModifiedFileInfoPtr == nil || fileInfo.modTime.After(lastModifiedFileInfoPtr.modTime) {
+						*lastModifiedFileInfoPtr = fileInfo
+					}
+				}
+			}
+
+			// return with the most strict match
+			if len(lastModifiedFileInfoPtr.path) > 0 {
+				return lastModifiedFileInfoPtr
+			}
+		}
+	} else {
+		for _, fileInfo := range fileInfos {
+			if lastModifiedFileInfoPtr == nil || fileInfo.modTime.After(lastModifiedFileInfoPtr.modTime) {
+				*lastModifiedFileInfoPtr = fileInfo
+			}
+		}
+	}
+
+	return lastModifiedFileInfoPtr
+}
+
+// exports the last modified file matching to most strict regexps within a time window
+// order of regexps should be: most strict -> less strict
+func findLastModifiedWithFileNameRegexpsInTimeWindow(dir string, startTime, endTime time.Time, patterns ...string) (string, error) {
```

Comment:
function name is 46 characters long. Something like "findBuildArtifacts" would be more appropriate.